src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.Stream.cs

```diff
@@ -50,52 +50,60 @@ private static async Task WriteAsyncCore(object value, Type type, Stream utf8Jso
                 options = JsonSerializerOptions.s_defaultOptions;
             }
 
-            JsonWriterOptions writerOptions = options.GetWriterOptions();
-
             using (var bufferWriter = new PooledBufferWriter<byte>(options.DefaultBufferSize))
-            using (var writer = new Utf8JsonWriter(bufferWriter, writerOptions))
             {
-                if (value == null)
+                var cachedWriter = CachedUtf8JsonWriter.Get(bufferWriter, options.GetWriterOptions());
```

Comment:
@steveharter, why is the local called `bufferWriter` here, but `output` in `WriteCore` in the helpers file? Either is fine, but we should try to be consistent.